api/src/main/java/org/openmrs/module/openconceptlab/OpenConceptLabActivator.java

```diff
@@ -41,10 +50,51 @@ public void contextRefreshed() {
 		if (!Context.isSessionOpen()) {
 			Context.openSession();
 		}
-			
+
+		//TODO: is this path OK?
+		String loadAtStartupPath = Context.getAdministrationService()
+				.getGlobalProperty(OpenConceptLabConstants.GP_OCL_LOAD_AT_STARTUP_PATH);
+		if (StringUtils.isBlank(loadAtStartupPath)) {
+			loadAtStartupPath = OpenmrsUtil.getApplicationDataDirectory()
+					+ (OpenmrsUtil.getApplicationDataDirectory().endsWith(File.separator) ? "ocl" : File.separator + "ocl")
+					+ File.separator + "loadAtStartup";
+			Context.getAdministrationService()
+					.saveGlobalProperty(
+							new GlobalProperty(OpenConceptLabConstants.GP_OCL_LOAD_AT_STARTUP_PATH, loadAtStartupPath)
+					);
+		}
+
+		File loadAtStartupDir = new File(loadAtStartupPath);
+		if (!loadAtStartupDir.exists()) {
+			loadAtStartupDir.mkdirs();
+		}
+
 		UpdateScheduler scheduler = Context.getRegisteredComponent("openconceptlab.updateScheduler", UpdateScheduler.class);
-		scheduler .scheduleUpdate();
-		
+
+		File[] files = loadAtStartupDir.listFiles();
+		if (files.length > 1) {
+			log.error("There is more than one file in ocl/loadAtStartup directory");
+			log.error("Ensure that there is only one file");
+			log.error("Absolute directory path: " + loadAtStartupDir.getAbsolutePath());
+		} else if (files.length != 0) {
+			File file = files[0];
+			if (file.getName().endsWith(".zip")) {
+				InputStream inputStream;
+				try {
+					inputStream = Utils.extractExportInputStreamFromZip(new ZipFile(file));
+					Importer importer = Context.getRegisteredComponent("openconceptlab.importer", Importer.class);
+					importer.setInputStream(inputStream);
+					scheduler.scheduleNow();
+					file.delete();
+				} catch (IOException e) {
+					log.error("Failed to open zip file", e);
+				}
+			} else {
+				log.error("File " + file.getName() + " must be in *.zip format");
+			}
```

Comment:
Instead of logging errors, please throw exceptions. They should prevent the module from starting up and the error will be better communicated to the sys admin.